CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -523,10 +523,11 @@ public function buildQuickForm() {
       $getTaxDetails = FALSE;
       $taxTerm = CRM_Utils_Array::value('tax_term', $invoiceSettings);
```

Comment:
So there doesn't seem to be an easy way of stripping zero decimals in smarty - there's things like this but it seems a bit heavy for a tpl: https://stackoverflow.com/questions/26565650/smarty-trim-zeros-from-the-end-of-string-float-value.  I have just pushed a simpler, less invasive version of the previous patch.  It still uses the cast to float as before, but only makes a single line change to Confirm.php (Thankyou.php requires a block of code to be moved so the lineItem variables are set for display but this is lower risk as Thankyou.php does not save anything, only displays to the end user).  I have run tests with multiple line items on the contribution pages and can find no issues with displaying the line items (both use LineItem.tpl for actual display).  @KarinG @monishdeb what do you think of the updated patch?